### ISSUE TITLE
Create GEF release 3.14.0 and contribute to simrel

### ISSUE THREAD

**Issue by vogella** [eclipse/gef-classic#40]
Now that GEF has been updated to Java 11, we should build a release which we could contribute to Simrel. As platform and SWT already require Java 11 I assume that will not a breaking change for anyone

This would require https://github.com/eclipse/gef-classic/issues/37 to be fixed first to add signing and an update of the build job.

**Comment by vogella** [comment]
Process for simrel is described here: https://wiki.eclipse.org/SimRel/Overview

**Comment by azoitl** [comment]
@vogella I think contributing the current state to simrel is a very good idea. At this point we mostly updated descriptions, the build system and updated to Java 11. So this shouldn't really break anything in clients. My first test we Eclipse 4diac was also very promising. I had a qick look at the SimRel docs. What I not fully got was is a contribution the built by simrel or is the aggregator taking a p2 repo with a release and integrates it into simrel?

But I think in both cases we would need some kind of official GEF classic release with a release update site or?

**Comment by pcdavid** [comment]
> So this shouldn't really break anything in clients.

The changes in feature IDs made in https://github.com/eclipse/gef-classic/commit/086615ed87137a007e403d9fb2065a8c477937bf *are* breaking. For example [this change](https://github.com/eclipse/gef-classic/commit/086615ed87137a007e403d9fb2065a8c477937bf#diff-716d79f77803c25a5870d8ef8313812bc5971573fe79349536dbb00036477b02L13) breaks [the main GMF Runtime feature](https://github.com/eclipse/gmf-runtime/blob/master/org.eclipse.gmf.runtime-feature/feature.xml#L38):

```
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.gmf.feature.group 1.14.2.qualifier
[ERROR]   Missing requirement: org.eclipse.gmf.feature.group 1.14.2.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.gef.feature.group 0.0.0' but it could not be found
```

Yes it's easily fixable, but requires a new GMF Runtime release (at least, maybe also in downstream projects) and I don't know when I will find the time to do it.

**Comment by vogella** [comment]
Ah, I think it is easier if we change it back.... Sorry for this.

**Comment by pcdavid** [comment]
Removing the feature is neither more not less work that updating the name. In both cases I need to make a new version.

I'll update the feature name in GMF, do not change it back. I need to make a new version anyway to move to Java 11.
Both changes are minor, it's just that I don't have much time to work on GMF, so I can not say *when* this will happen.

When do you plan to contribute GEF Classic to the SimRel? M2?

**Comment by vogella** [comment]
> When do you plan to contribute GEF Classic to the SimRel? M2?

M2 would be great, if that would work for you.

**Comment by mickaelistria** [comment]
> @mickaelistria suggested in a bug in platform to remove the required feature in your own feature 

Not really, I suggested that for bundles that are already required transtively by some Required-Bundle or Import-Package. As there is no other way to specific feature requirement that requirements in feature.xml, then there is no strong reason to remove those feature requirements if they're relevant to the project.
For example, a project that adds features to JDT debug model will transtively require JDT debug, but may want its feature to require the whole JDT feature because it contains debug.ui and other stuff that are relevant for the project features although they're not transitive dependencies of project's bundles.

> Removing the feature is neither more not less work that updating the name. 

Removing is a 1-time work, maintaining name is a "whenever this name change" effort.

**Comment by vogella** [comment]
Checklist from WB https://github.com/eclipse/windowbuilder/wiki/Release-Checklist

Thanks @wimjongman

**Comment by azoitl** [comment]
This checklist is very nice. Thanks @wimjongman! 

I noticed that I have to recheck all versions a second time. I think one of the features has an odd version. Will do that today.

**Comment by vogella** [comment]
https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/264

**Comment by pcdavid** [comment]
> > When do you plan to contribute GEF Classic to the SimRel? M2?
> 
> M2 would be great, if that would work for you.

I've just published an RC1 for GMF Runtime 1.15.0 which switches to GEF Classic (and Java 11): https://download.eclipse.org/modeling/gmp/gmf-runtime/updates/milestones/S202205020810/

It's building against https://download.eclipse.org/tools/gef/classic/latest (currently a version built on 2022-04-22). I'd rather point to a fixed version (at least for the final build) when you have one available.

The patch to contribute this to the SimRel is [here](https://git.eclipse.org/r/c/simrel/org.eclipse.simrel.build/+/193032), but currently fails because it now depends on GEF Classic. If/when you have a build to contribute to the SimRel could you put your patch on top of this one so they can be merged together as they are interdependant?

**Comment by vogella** [comment]
Awesome. Our release review is scheduled for 11.05, the next day we can contribute.

**Comment by vogella** [comment]
GEF classic release number will be "3.11.100" unless someone disagrees.

**Comment by wimjongman** [comment]
Why 100 and not just 3.11.0 or 3.12.0? 100 looks like a patch version to me.

**Comment by vogella** [comment]
Features have this number.

**Comment by vogella** [comment]
Aktually I see a mix of numbers in the features. I think we should allign them and move them to 3.13.0 to indicate a real, non breaking release.

**Comment by azoitl** [comment]
the GEF classic features had 3.11.xxx in the last release. I upgraded all minor version numers and set the last number (never now how it is called) to 100 as suggested by @vogella. So the release should be 3.12.0, or?

**Comment by vogella** [comment]
@azoitl we never can go back in the versions, e.g., if we now move from 3.12.100 to 3.12.0 consumer like GMF would not pick up the new version.

I aligned everything around 3.13.0, here is  picture of the local update site.

![image](https://user-images.githubusercontent.com/139910/166901617-0ee4a34d-afaf-45ce-bf0c-8a124fb7fc53.png)

**Comment by wimjongman** [comment]
>(never know how it is called)

major.minor.patch

> I aligned everything around 3.13.0, here is picture of the local update site.

13 😨  😄

**Comment by vogella** [comment]
@pcdavid and @azoitl please sanity check http://download.eclipse.org/tools/gef/classic/releases/3.13.0 which I plan to contribute to the simrel tomorrow.

**Comment by vogella** [comment]
> 13 fearful smile

Lucky number somewhere in the world. :-)

**Comment by azoitl** [comment]
> @azoitl we never can go back in the versions, e.g., if we now move from 3.12.100 to 3.12.0 consumer like GMF would not pick up the new version.
That is a good point. And putting everything to the same release version makes life much easier for writing documenation, changelogs, etc.

**Comment by vogella** [comment]
Contribution to simrel https://git.eclipse.org/r/c/simrel/org.eclipse.simrel.build/+/193185

**Comment by pcdavid** [comment]
I've made an RC2 build of GMF Runtime 1.15.0 at https://download.eclipse.org/modeling/gmp/gmf-runtime/updates/milestones/S202205071245/.
Trying to contribute that to the SimRel (on top of your commit) reveals that DTP (Data Tools) also depends on GEF and like GMF references the feature whose id has changed:

```
Build failed! Exception was org.eclipse.core.runtime.CoreException: Cannot complete the install because one or more required items could not be found. Software being installed: validationSet_main 1.0.0 Missing requirement: org.eclipse.datatools.sqldevtools.feature.feature.group 1.14.201.202107090755 requires 'org.eclipse.equinox.p2.iu; org.eclipse.gef.feature.group 3.4.0' but it could not be found Cannot satisfy dependency: mappedRepo_download.eclipse.org_datatools_updates_1.14.300-SNAPSHOT_repository 1.0.0 depends on: org.eclipse.equinox.p2.iu; org.eclipse.datatools.sqldevtools.feature.feature.group 0.0.0 Cannot satisfy dependency: validationSet_main 1.0.0 depends on: org.eclipse.equinox.p2.iu; mappedRepo_download.eclipse.org_datatools_updates_1.14.300-SNAPSHOT_repository [1.0.0,1.0.0]
```

I tried [disabling it](https://git.eclipse.org/r/c/simrel/org.eclipse.simrel.build/+/193209), but WebTools depends on it:

```
16:09:58 Missing requirement: Dali Java Persistence Tools - JPA DB 2.2.201.v202106200606 (org.eclipse.jpt.jpa.db 2.2.201.v202106200606) requires 'osgi.bundle; org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)' but it could not be found
16:09:58 
16:09:58 Bundle(org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)) is required by:
16:09:58   ValidationSet(main)
16:09:58     Contribution(Web Tools Platform 3.26 for SimRel 2022-06)
16:09:58       MappedRepository(https://download.eclipse.org/webtools/downloads/drops/R3.26.0/S-3.26M1a-20220411232710/repository/)
16:09:58         Feature(org.eclipse.jpt.jpa.eclipselink.feature.feature.group)
16:09:58           InstallableUnit(org.eclipse.jpt.jpa.eclipselink.core 2.4.101.v201902121834)
16:09:58             InstallableUnit(org.eclipse.jpt.jpa.core 3.6.1.v202106020926)
16:09:58               InstallableUnit(org.eclipse.jpt.jpa.db 2.2.201.v202106200606)
16:09:58 Done. Took 3 seconds
16:09:58 Build failed! Exception was org.eclipse.core.runtime.CoreException: Cannot complete the install because one or more required items could not be found. Software being installed: validationSet_main 1.0.0 Missing requirement: org.eclipse.jpt.jpa.db 2.2.201.v202106200606 requires 'osgi.bundle; org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)' but it could not be found Cannot satisfy dependency: mappedRepo_download.eclipse.org_webtools_downloads_drops_R3.26.0_S-3.26M1a-20220411232710_repository 1.0.0 depends on: org.eclipse.equinox.p2.iu; org.eclipse.jpt.jpa.eclipselink.feature.feature.group 0.0.0 Cannot satisfy dependency: org.eclipse.jpt.jpa.core 3.6.1.v202106020926 depends on: osgi.bundle; org.eclipse.jpt.jpa.db [2.0.0,3.0.0) Cannot satisfy dependency: org.eclipse.jpt.jpa.eclipselink.core 2.4.101.v201902121834 depends on: osgi.bundle; org.eclipse.jpt.jpa.core [3.0.0,4.0.0) Cannot satisfy dependency: org.eclipse.jpt.jpa.eclipselink.feature.feature.group 3.4.101.v201903221940 depends on: org.eclipse.equinox.p2.iu; org.eclipse.jpt.jpa.eclipselink.core [2.4.101.v201902121834,2.4.101.v201902121834] Cannot satisfy dependency: validationSet_main 1.0.0 depends on: org.eclipse.equinox.p2.iu; mappedRepo_download.eclipse.org_webtools_downloads_drops_R3.26.0_S-3.26M1a-20220411232710_repository [1.0.0,1.0.0]
16:09:58 Cannot complete the install because one or more required items could not be found. Software being installed: validationSet_main 1.0.0 Missing requirement: org.eclipse.jpt.jpa.db 2.2.201.v202106200606 requires 'osgi.bundle; org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)' but it could not be found Cannot satisfy dependency: mappedRepo_download.eclipse.org_webtools_downloads_drops_R3.26.0_S-3.26M1a-20220411232710_repository 1.0.0 depends on: org.eclipse.equinox.p2.iu; org.eclipse.jpt.jpa.eclipselink.feature.feature.group 0.0.0 Cannot satisfy dependency: org.eclipse.jpt.jpa.core 3.6.1.v202106020926 depends on: osgi.bundle; org.eclipse.jpt.jpa.db [2.0.0,3.0.0) Cannot satisfy dependency: org.eclipse.jpt.jpa.eclipselink.core 2.4.101.v201902121834 depends on: osgi.bundle; org.eclipse.jpt.jpa.core [3.0.0,4.0.0) Cannot satisfy dependency: org.eclipse.jpt.jpa.eclipselink.feature.feature.group 3.4.101.v201903221940 depends on: org.eclipse.equinox.p2.iu; org.eclipse.jpt.jpa.eclipselink.core [2.4.101.v201902121834,2.4.101.v201902121834] Cannot satisfy dependency: validationSet_main 1.0.0 depends on: org.eclipse.equinox.p2.iu; mappedRepo_download.eclipse.org_webtools_downloads_drops_R3.26.0_S-3.26M1a-20220411232710_repository [1.0.0,1.0.0]
16:09:58 [0]Missing requirement: Dali Java Persistence Tools - JPA DB 2.2.201.v202106200606 (org.eclipse.jpt.jpa.db 2.2.201.v202106200606) requires 'osgi.bundle; org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)' but it could not be found
16:09:58 
16:09:58 Bundle(org.eclipse.datatools.enablement.jdt.classpath [1.0.1,3.0.0)) is required by:
16:09:58   ValidationSet(main)
16:09:58     Contribution(Web Tools Platform 3.26 for SimRel 2022-06)
16:09:58       MappedRepository(https://download.eclipse.org/webtools/downloads/drops/R3.26.0/S-3.26M1a-20220411232710/repository/)
16:09:58         Feature(org.eclipse.jpt.jpa.eclipselink.feature.feature.group)
16:09:58           InstallableUnit(org.eclipse.jpt.jpa.eclipselink.core 2.4.101.v201902121834)
16:09:58             InstallableUnit(org.eclipse.jpt.jpa.core 3.6.1.v202106020926)
16:09:58               InstallableUnit(org.eclipse.jpt.jpa.db 2.2.201.v202106200606)
```

**Comment by pcdavid** [comment]
I had not noticed earlier, but it seems the new GEF Class repos do not include the sources?

Compare:
* https://download.eclipse.org/tools/gef/classic/releases/3.13.0
* https://download.eclipse.org/tools/gef/updates/legacy/releases/4.0.0_gef-master_1952

That's quite annoying when working on downstream projects.

**Comment by pcdavid** [comment]
I guess this one could be closed? GEF Classic 3.14 has been released and contributed to the SimRel post-2022-06.
